### PR TITLE
Add unit integration and e2e tests

### PR DESCRIPTION
**Pull request by myroslavve**
Added:
Unit tests:
White box - `src/app/_components/tracks/hooks/__tests__/useQueryParams.test.ts`
Black box - `src/lib/utils/__tests__/utils.test.ts`
Integration tests:
`src/components/upload-modal/__tests__/index.test.tsx`
E2E tests:
`e2e-tests/app/components/tracks.test.ts`

### CODE REVIEW

File: src/lib/utils/__tests__/utils.test.ts
Comment (igaryakqwe) [inline src/lib/utils/__tests__/utils.test.ts:1]: If it's blackbox testing, there shouldn't be any mocks

**Review by igaryakqwe — CHANGES_REQUESTED** [review]
I didn't found integration tests on logic

**Comment by myroslavve** [comment]
> I haven't found integration tests on logic

Added one in `src/lib/api/__tests__/genres.test.ts`

File: src/lib/utils/__tests__/utils.test.ts
Comment (myroslavve) [reply src/lib/utils/__tests__/utils.test.ts:1]: I'm not mocking internals of the function, I'm passing a mock to it just to spy on whether the mock was called. I could implement this without mock function by making a function that sets a variable like `wasCalled` to true when called or by using `vi.spy` on a function, but the mock does it for me so it's more convenient IMO.
So I would argue that this is still a blackbox test.

File: playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// import dotenv from 'dotenv';
+// import path from 'path';
+// dotenv.config({ path: path.resolve(__dirname, '.env') });
+
+/**
+ * See https://playwright.dev/docs/test-configuration.
+ */
```
Comment (AntonZaiets) [inline playwright.config.ts:13]: Do we need commented parts of the code?

Comment (myroslavve) [reply playwright.config.ts]: I don't think those are needed, removed them. Thanks 👍

**Review by igaryakqwe — APPROVED** [review]

File: src/app/_components/tracks/hooks/__tests__/useQueryParams.test.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        const mockParams = {
+            param1: Belt.O.Some('value1'),
+            param2: Belt.O.Some('value2')
+        }
+        
+        mockCreateParamsObject.mockReturnValue(mockParams)
+        
+        const { result } = renderHook(() => useQueryParams(paramsList))
+        
+        // Call setParam multiple times
```
Comment (GluterusMaximus) [inline src/app/_components/tracks/hooks/__tests__/useQueryParams.test.ts:163]: Do we need this comment here? Code seems to be self-explanatory

File: src/components/upload-modal/__tests__/UploadModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import { describe, it, expect, vi } from 'vitest'
+import UploadModal from '..'
+import { Button } from '../../ui/button'
+
+describe('UploadModal', () => {
```
Comment (GluterusMaximus) [inline src/components/upload-modal/__tests__/UploadModal.test.tsx:6]: Do we need the describe here? It is one per file, has the same name as the file and doesn't have any side effects / before / after hooks

File: src/components/upload-modal/__tests__/UploadModal.test.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    render(<UploadModal onUpload={() => {}} onDelete={() => {}} uploaded="" />)
+    
+    fireEvent.click(screen.getByText('Upload'))
+    
+    const file = new File(['audio content'], 'test.mp3', { type: 'audio/mpeg' })
+    const input = screen.getByTestId('file-input')
+    
+    fireEvent.change(input, { target: { files: [file] } })
+    
+    expect(screen.getByText('Selected: test.mp3')).toBeDefined()
```
Comment (GluterusMaximus) [inline src/components/upload-modal/__tests__/UploadModal.test.tsx:37]: Just an idea, maybe check that there are no validation errors here?

File: src/components/upload-modal/__tests__/UploadModal.test.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    fireEvent.click(screen.getByText('Upload'))
+
+    const file = new File(['audio content'], 'test.txt', { type: 'text/plain' })
+    const input = screen.getByTestId('file-input')
+
+    fireEvent.change(input, { target: { files: [file] } })
+    fireEvent.click(screen.getByTestId('upload-button'))
+
+    await expect(screen.findByText('File must be a valid MP3 or WAV file')).resolves.toBeDefined()
```
Comment (GluterusMaximus) [inline src/components/upload-modal/__tests__/UploadModal.test.tsx:51]: Might be a good idea to check, that the file was not selected

File: src/lib/api/__tests__/genres.test.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+      expect(Array.isArray(result.value)).toBe(true);
+      expect(result.value.length).toBeGreaterThan(0);
+    }
+  });
+
+  it('should return cached genres on subsequent calls', async () => {
+    const firstResult = await getGenres();
+    expect(firstResult.isOk()).toBe(true);
+    
+    // Mock the server to return different data
```
Comment (GluterusMaximus) [inline src/lib/api/__tests__/genres.test.ts:58]: Do we need comments here?

File: src/lib/api/__tests__/genres.test.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+afterAll(() => {
+  server.close();
+});
+
+describe('getGenres Integration Tests', () => {
+  it('should successfully fetch and return genres from the API', async () => {
+    const result = await getGenres();
+    
+    expect(result.isOk()).toBe(true);
+    if (result.isOk()) {
```
Comment (GluterusMaximus) [inline src/lib/api/__tests__/genres.test.ts:40]: Do we need an if statement here? Wouldn't the test fail anyway, if result.isOk is not truthy?

Same for other tests

File: src/lib/api/__tests__/genres.test.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        return HttpResponse.json([]);
+      })
+    );
+
+    const result = await getGenres();
+    
+    expect(result.isOk()).toBe(true);
+    if (result.isOk()) {
+      expect(result.value).toEqual([]);
+      expect(Array.isArray(result.value)).toBe(true);
```
Comment (GluterusMaximus) [inline src/lib/api/__tests__/genres.test.ts:134]: This check seems redundant - we already checked that we have an empty array on the previous line.

**Review by GluterusMaximus — CHANGES_REQUESTED** [review]
Overall - good job testing different corner cases and error states 🔥

File: src/components/upload-modal/__tests__/UploadModal.test.tsx (hunk 3 above)
Comment (myroslavve) [reply src/components/upload-modal/__tests__/UploadModal.test.tsx]: I like that it shows what we're testing in a normal text, not file path, but sure, isn't necessary here, removed.

File: src/components/upload-modal/__tests__/UploadModal.test.tsx (hunk 4 above)
Comment (myroslavve) [reply src/components/upload-modal/__tests__/UploadModal.test.tsx]: Implemented 👍

File: src/components/upload-modal/__tests__/UploadModal.test.tsx (hunk 5 above)
Comment (myroslavve) [reply src/components/upload-modal/__tests__/UploadModal.test.tsx]: The validation happens only when user clicks "Upload" button, so the file is actually selected here. Don't want to change how it works, in this PR at least, so I didn't add the check.

**Review by GluterusMaximus — APPROVED** [review]
